pkg/reconciler/config_reconciler.go

```diff
@@ -0,0 +1,192 @@
+package reconciler
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/telekom/das-schiff-network-operator/api/v1alpha1"
+	"github.com/telekom/das-schiff-network-operator/pkg/debounce"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	DefaultTimeout         = "60s"
+	DefaultNodeUpdateLimit = 1
+)
+
+// ConfigReconciler is responsible for creating NodeConfig objects.
+type ConfigReconciler struct {
+	logger    logr.Logger
+	debouncer *debounce.Debouncer
+	client    client.Client
+	timeout   time.Duration
+}
+
+type reconcileConfig struct {
+	*ConfigReconciler
+	logr.Logger
+}
+
+// Reconcile starts reconciliation.
+func (cr *ConfigReconciler) Reconcile(ctx context.Context) {
+	cr.debouncer.Debounce(ctx)
+}
+
+// // NewConfigReconciler creates new reconciler that creates NetworkConfigRevision objects.
+func NewConfigReconciler(clusterClient client.Client, logger logr.Logger, timeout time.Duration) (*ConfigReconciler, error) {
+	reconciler := &ConfigReconciler{
+		logger:  logger,
+		timeout: timeout,
+		client:  clusterClient,
+	}
+
+	reconciler.debouncer = debounce.NewDebouncer(reconciler.ReconcileDebounced, defaultDebounceTime, logger)
+
+	return reconciler, nil
+}
+
+func (cr *ConfigReconciler) ReconcileDebounced(ctx context.Context) error {
+	r := &reconcileConfig{
+		ConfigReconciler: cr,
+		Logger:           cr.logger,
+	}
+
+	cr.logger.Info("fetching config data...")
+
+	timeoutCtx, cancel := context.WithTimeout(ctx, cr.timeout)
+	defer cancel()
+
+	// get VRFRouteConfiguration, Layer2networkConfiguration and RoutingTable objects
+	configData, err := r.fetchConfigData(timeoutCtx)
+	if err != nil {
+		return fmt.Errorf("error fetching configuration details: %w", err)
+	}
+
+	// prepare new revision
+	revision, err := v1alpha1.NewRevision(configData)
+	if err != nil {
+		return fmt.Errorf("error preparing new config revision: %w", err)
+	}
+
+	cr.logger.Info("new revision", "data", revision)
+
+	// get all known revisions
+	revisions, err := listRevisions(timeoutCtx, cr.client)
+	if err != nil {
+		return fmt.Errorf("error listing revisions: %w", err)
+	}
+
+	// check if revision should be skipped (e.g. it is the same as known invalid revision, or as currently deployed revision)
+	if shouldSkip(revisions, revision) {
```

Comment:
can we log why we've skipped a revision? Otherwise it might be hard to figure out

Reply:
Done